### PR TITLE
Added code for Sum of Digit

### LINKED ISSUE #2726: To find the Sum of Digits
## 💥 Proposal

The code will sum up the individual digits of the number input by user.

"Example"

1) Suppose the user inputs 345, the output will be 3+4+5=12;
2) The user inputs 8975,the output will be 8+9+7+5=29;

Hey @hushee69 , I want to add this question in the 'NeoAlgo/Java/math' as I didn't find it the java folder. So, I want to contribute the same as a part of GSSOC'21.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

YES

### PR DESCRIPTION
**Pull request by Sweeti-Gupta**
Fixed #2726

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
-YES

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I have added the code for  calculating the sum of digits of a number and fixed the issue- #2726 . Please review and merge it.

### CODE REVIEW

File: Java/math/SumOfDigit.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		return sum;
+	}
+	
+	public static void main(String[] args)
+	{
+		System.out.println("Enter the number for which you want to calculate the sum of digits:");
+		Scanner sc = new Scanner(System.in);
+		int n = sc.nextInt();
+		int x = getSum(n);
+		System.out.println(x);
```
Comment (atarax665) [inline Java/math/SumOfDigit.java]: Format your output as ```Sum of the digits of X is Y```.

File: Java/math/SumOfDigit.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		
+	}
+}
+
+
```
Comment (atarax665) [inline Java/math/SumOfDigit.java]: Remove all extra blank lines.

File: Java/math/SumOfDigit.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Example 2:
+
+Input:
+Enter the number for which you want to calculate the sum of digits:
+8674
+Output: 25
+Explanation: Sum of digits of the number 8674 = (8+6+7+4) = 25
+
```
Comment (atarax665) [inline Java/math/SumOfDigit.java:48]: Add space and time complexity of the program.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Sweeti-Gupta** [comment]
Hey @atarax665, I have made all the necessary changes. Please review and merge it.

**Comment by Manish5237** [comment]
Hi ,

I am GSSOC'21 participant.
Please assign me this. I will do this in C with proper comments and test cases.

@atarax665

**Comment by atarax665** [comment]
> Hi ,
> 
> I am GSSOC'21 participant.
> Please assign me this. I will do this in C with proper comments and test cases.
> 
> @atarax665

Commented on the linked issue #2726  to get it assigned to you.

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Pull request merged**